Entity/Transaction.php

```diff
@@ -459,17 +459,17 @@ public function getSuccess()
      */
     public function setMessages(ArrayCollection $messages = null)
     {
-        $this->messages = $messages->isEmpty() ? null : $messages;
+        $this->messages = empty($messages) || $messages->isEmpty() ? null : $messages->toArray();
```

Comment:
The problem this was trying to address is that doctrine does not store php serialized data into sql server when its configured this way (it throws an error which is assumed to be caused by mis-handled character escaping in doctrine for sql server). @RuslanZavacky and I discussed a) storing this as a regular php array (this won't work because it could be an array of objects) b) base64 encoding (this would make it impractical for any non-software system to decode and translate, or c) having these object be encoded into json before being stored in the db.  We decided we should explore option c) because json is more translatable to many db systems.  Lets investigate of the json_array type in doctrine will work with sql server, otherwise we can just hardcode a json_encode and decode for set and get message methods.  As for null, there is no reason null needs to be allowed as a default argument value, so that can be removed.  If a person wants to empty already set messages, they can send an empty ArrayCollection.  http://doctrine-dbal.readthedocs.org/en/latest/reference/types.html